llvm/lib/IR/Mangler.cpp

```diff
@@ -18,8 +18,10 @@
 #include "llvm/IR/DerivedTypes.h"
 #include "llvm/IR/Function.h"
 #include "llvm/IR/Module.h"
+#include "llvm/Object/COFF.h"
```

Comment:
(this comes up when I tried to fix the upstream bazel build [failure](https://buildkite.com/llvm-project/upstream-bazel/builds/93424))

We seem to introduce a cycle dependency here, in bazel config, we have
- llvm:Core target which includes all lib/IR/* files
- llvm:Object target depends on llvm:Core

This change makes llvm:Core depends on llvm:Object. Can we remove this dependency? We only need `object::getArm64ECDemangledFunctionName` in this file, is it possible (and sensible) to move this inline `getArm64ECDemangledFunctionName` from `COFF.h` to `Mangler.h` or a new file?